File: Data Structures & Algorithms/top-k-elements-in-list/submission-0.py
class Solution:
    def topKFrequent(self, nums: List[int], k: int) -> List[int]:
        cache = {}
        for i in nums:
            if i in cache.keys():
                cache[i] +=1
            else:
                cache[i] = 1

        reduced = [[key, val] for key, val in cache.items()]

        reduced = sorted(reduced, key=lambda x: x[1], reverse=True)

        result = []
        for i in reduced:
            result.append(i[0])
            if len(result) == k:
                break
        
        return result

File: Data Structures & Algorithms/products-of-array-discluding-self/submission-0.cpp
class Solution {
public:
    vector<int> productExceptSelf(vector<int>& nums) {
        int siz = nums.size();
        vector<int> result(siz, 1);
        int prefix = 1;
        for(int i=0;i< siz;i++){
            result[i] = result[i] * prefix;
            prefix = prefix * nums[i];
            std::cout << prefix << std::endl;
        }

        int postfix = 1;
        for(int i=siz-1;i>=0;i--){
            result[i] = result[i] * postfix;
            postfix = postfix * nums[i];
            std::cout << postfix << std::endl;
        }
        return result;
    }
};
